spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationprocessor/ConfigurationMetadataAnnotationProcessorTests.java

```diff
@@ -112,8 +112,11 @@ void simpleProperties() {
 			.fromSource(SimpleProperties.class)
 			.withDescription("A simple flag.")
 			.withDeprecation(null, null));
+		assertThat(metadata).has(Metadata.withProperty("simple.number", Long.class)
+				.fromSource(SimpleProperties.class)
+				.withDescription("A setter-only property.")
+		);
```

Comment:
I have renamed the property, a "counter", I think of a number of requests processed, most likely should not be manipulated from the outside, but a "number", I think of a concurrency limit, is legal and useful to manipulate.